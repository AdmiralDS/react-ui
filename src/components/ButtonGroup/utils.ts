export const moveFocus = (
  parent: HTMLDivElement | null,
  currentFocus: Element | null,
  calcNextFocus: (parent: HTMLDivElement | null, child: Element | null) => any,
): void => {
  let wrappedOnce = false;
  let nextFocus = calcNextFocus(parent, currentFocus);
  while (nextFocus) {
    if (nextFocus === parent?.firstChild) {
      if (wrappedOnce) {
        return;
      }
      wrappedOnce = true;
    }

    if (nextFocus.hasAttribute('disabled')) {
      nextFocus = calcNextFocus(parent, nextFocus);
    } else {
      nextFocus.focus();
      return;
    }
  }
};

export const nextItem = (parent: Element | null, child: Element | null): Element | null | undefined | ChildNode => {
  if (child?.nextElementSibling) {
    return child.nextElementSibling;
  }
  return parent?.firstChild;
};

export const previousItem = (parent: Element | null, child: Element | null): Element | null | undefined | ChildNode => {
  if (child?.previousElementSibling) {
    return child.previousElementSibling;
  }
  return parent?.lastChild;
};
