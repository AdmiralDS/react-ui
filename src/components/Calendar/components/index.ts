export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
