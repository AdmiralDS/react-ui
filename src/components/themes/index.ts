export * from './light';
export * from './dark';
export type { Theme } from './common';
