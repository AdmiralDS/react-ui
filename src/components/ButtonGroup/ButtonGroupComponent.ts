import { HTMLAttributes } from 'react';
import styled from 'styled-components';
import { ButtonGroupItem } from './ButtonGroupItem';

export type Dimension = 'l' | 'm' | 's';

export const SIZE = {
  l: 40,
  m: 32,
  s: 24,
};

type ButtonGroupComponentProps = HTMLAttributes<HTMLDivElement> & {
  dimension: Dimension;
};

const BORDER_RADIUS = {
  l: 24,
  m: 20,
  s: 16,
};

const padding = {
  l: 20,
  m: 16,
  s: 12,
};

export const ButtonGroupComponent = styled.div<ButtonGroupComponentProps>`
  background: ${({ theme }) => theme.color.background.tertiary};
  border: 2px solid ${(p) => p.theme.color.background.tertiary};
  border-radius: ${({ dimension }) => BORDER_RADIUS[dimension]}px;
  display: inline-flex;
  height: ${({ dimension }) => SIZE[dimension]}px;
  padding: 2px;

  &[data-appearance~='primary'] {
    border: 2px solid ${(p) => p.theme.color.basic.primary};
    background-color: transparent;
  }

  ${ButtonGroupItem} {
    height: 100%;
    padding: ${({ dimension }) => `0 ${padding[dimension]}px`};
    font-size: ${({ dimension }) => (dimension === 'l' ? 16 : 14)}px;
    line-height: ${({ dimension }) => (dimension === 'l' ? 24 : 20)}px;
  }

  > *:not(:first-child) {
    margin-left: 8px;
  }
`;
