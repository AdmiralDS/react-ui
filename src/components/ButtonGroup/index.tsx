import React, { HTMLAttributes, KeyboardEvent, ReactNode, useRef } from 'react';
import type { Dimension } from './ButtonGroupComponent';
import { ButtonGroupComponent } from './ButtonGroupComponent';
import { moveFocus, nextItem, previousItem } from './utils';
import { keyboardKey } from '#src/components/common/keyboardKey';

export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement> {
  /** Размер компонента */
  dimension?: Dimension;
  /** Элементы содержимого */
  children?: ReactNode;
  /** Стиль отображения компонента */
  appearance?: 'primary' | 'secondary';
}

export const ButtonGroup = ({
  dimension = 'l',
  tabIndex = -1,
  appearance = 'secondary',
  ...props
}: ButtonGroupProps) => {
  const localRef = useRef<HTMLDivElement>(null);

  const getFocusedOption = () => {
    return ((localRef.current && localRef.current.ownerDocument) || document).activeElement;
  };

  const handleKeyDown = (e: KeyboardEvent<HTMLDivElement>) => {
    const focusedOption = getFocusedOption();
    const code = keyboardKey.getCode(e);
    if (code === keyboardKey.ArrowRight) {
      moveFocus(localRef.current, focusedOption, nextItem);
    } else if (code === keyboardKey.ArrowLeft) {
      moveFocus(localRef.current, focusedOption, previousItem);
    }
    props?.onKeyDown?.(e);
  };

  return (
    <ButtonGroupComponent
      ref={localRef}
      onKeyDown={handleKeyDown}
      dimension={dimension}
      tabIndex={tabIndex}
      data-appearance={appearance}
      {...props}
    />
  );
};

ButtonGroup.displayName = 'ButtonGroup';
