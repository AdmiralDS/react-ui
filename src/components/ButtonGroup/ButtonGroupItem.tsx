import styled, { css } from 'styled-components';
import { typography } from '#src/components/Typography';
import { BadgeComponent } from '#src/components/Badge';

export type ButtonGroupItemProps = {
  /** Активная кнопка */
  active?: boolean;
};

const colorMixin = css<{ active?: boolean; disabled?: boolean }>`
  color: ${({ active, disabled, theme }) => {
    if (active) return theme.color.text.primary;
    if (disabled) return theme.color.text.tertiary;
    return theme.color.text.secondary;
  }};
  [data-appearance~='primary'] && {
    color: ${({ active, disabled, theme }) => {
      if (active) return theme.color.text.staticWhite;
      if (disabled) return theme.color.text.tertiary;
      return theme.color.basic.primary;
    }};
  }
`;

const BORDER_RADIUS = 22;

export const ButtonGroupItem = styled.button<ButtonGroupItemProps>`
  border: none;
  border-radius: ${BORDER_RADIUS}px;
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  display: flex;
  align-items: center;
  position: relative;
  outline: 0;
  background-color: ${({ theme, disabled, active }) =>
    active && !disabled ? theme.color.background.primary : 'transparent'};

  ${typography['Button/Button 1']};
  white-space: nowrap;

  ${colorMixin};

  &:hover:not(:disabled) {
    background: ${({ active, disabled, theme }) => (!active && !disabled ? theme.color.background.tertiaryHover : '')};
    color: ${(p) => p.theme.color.text.secondary};
  }

  & *[fill^='#'] {
    fill: ${({ disabled, theme }) => {
      if (disabled) return theme.color.text.tertiary;
      return theme.color.text.secondary;
    }};
  }

  [data-appearance~='primary'] && *[fill^='#'] {
    fill: ${({ active, disabled, theme }) => {
      if (active) return theme.color.text.staticWhite;
      if (disabled) return theme.color.text.tertiary;
      return theme.color.basic.primary;
    }};
  }

  & ${BadgeComponent} {
    background-color: ${(p) => (p.active ? p.theme.color.text.secondary : p.theme.color.background.primary)};
    color: ${(p) => (p.active ? p.theme.color.text.staticWhite : p.theme.color.text.secondary)};
  }

  [data-appearance~='primary'] && ${BadgeComponent} {
    background-color: ${({ active, disabled, theme }) => {
      if (active) return theme.color.text.staticWhite;
      if (disabled) return theme.color.text.tertiary;
      return theme.color.basic.primary;
    }};
    color: ${({ theme, disabled, active }) =>
      active && !disabled ? theme.color.basic.primary : theme.color.text.staticWhite};
  }

  [data-appearance~='primary'] && {
    background-color: ${({ theme, disabled, active }) =>
      active && !disabled ? theme.color.basic.primary : 'transparent'};
    &:hover:not(:disabled) {
      background: ${({ active, disabled, theme }) => (!active && !disabled ? theme.color.background.secondary : '')};
      color: ${(p) => (!p.active ? p.theme.color.basic.hover : p.theme.color.text.staticWhite)};
    }
  }
`;

ButtonGroupItem.defaultProps = {
  type: 'button',
};
