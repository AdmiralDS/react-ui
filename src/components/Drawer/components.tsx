import type { FC, HTMLAttributes } from 'react';
import { createContext, useContext, useLayoutEffect, useRef } from 'react';
import styled from 'styled-components';

import { typography } from '#src/components/Typography';
import { debounce } from '#src/components/common/utils/debounce';
import { resizePaddings } from '../common/dom/resizePaddings';

const Title = styled.h5<{ $mobile: boolean; $displayCloseIcon: boolean }>`
  ${({ $mobile }) => ($mobile ? typography['Header/H6'] : typography['Header/H5'])};
  color: var(--admiral-color-Neutral_Neutral90, ${(p) => p.theme.color['Neutral/Neutral 90']});
  margin: 0;
  padding: ${({ $mobile, $displayCloseIcon }) => {
    if ($mobile) {
      return $displayCloseIcon ? '0 48px 8px 16px' : '0 16px 8px';
    }
    return $displayCloseIcon ? '0 56px 8px 24px' : '0 24px 8px';
  }};
`;

const Content = styled.div<{ $mobile: boolean }>`
  overflow-y: auto;
  outline: none;
  scrollbar-gutter: stable;
  padding-block: 8px;
  padding-inline: ${({ $mobile }) => `${$mobile ? 16 : 24}px`};
  height: 100%;
`;

const ButtonPanel = styled.div<{ $mobile: boolean }>`
  display: flex;
  flex-direction: ${({ $mobile }) => ($mobile ? 'column-reverse' : 'row-reverse')};
  padding: ${({ $mobile }) => ($mobile ? '16px 16px 0' : '16px 24px 0')};

  & > button {
    margin: ${({ $mobile }) => ($mobile ? '0 0 16px 0' : '0 16px 0 0')};
    ${({ $mobile }) => $mobile && 'width: 100%;'}
  }

  & > button:first-child {
    margin: 0;
  }
`;

export const DrawerContext = createContext({ mobile: false, displayCloseIcon: true } as {
  mobile: boolean;
  displayCloseIcon: boolean;
});

export const DrawerTitle: FC<HTMLAttributes<HTMLHeadingElement>> = ({ children, ...props }) => {
  const { mobile, displayCloseIcon } = useContext(DrawerContext);
  const asProp = mobile ? 'h6' : 'h5';
  return (
    <Title $mobile={mobile} $displayCloseIcon={displayCloseIcon} as={asProp} {...props}>
      {children}
    </Title>
  );
};

export const DrawerContent: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const contentRef = useRef<HTMLDivElement | null>(null);
  const mobile = useContext(DrawerContext).mobile;

  useLayoutEffect(() => {
    const node = contentRef.current;
    if (node) {
      resizePaddings(node);
      const resizeObserver = new ResizeObserver(debounce(() => resizePaddings(node), 250));
      resizeObserver.observe(node);
      return () => {
        resizeObserver.unobserve(node);
      };
    }
  }, []);

  return (
    <Content tabIndex={-1} ref={contentRef} $mobile={mobile} {...props}>
      {children}
    </Content>
  );
};

export const DrawerButtonPanel: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const mobile = useContext(DrawerContext).mobile;
  return (
    <ButtonPanel $mobile={mobile} {...props}>
      {children}
    </ButtonPanel>
  );
};
