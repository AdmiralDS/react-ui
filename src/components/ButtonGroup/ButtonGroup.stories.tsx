import React, { useState } from 'react';
import styled from 'styled-components';
import { ComponentMeta, ComponentStory, Story } from '@storybook/react';
import { ButtonGroup } from '../ButtonGroup';
import { ButtonGroupItem } from '#/components/ButtonGroup/ButtonGroupItem';
import { ReactComponent as StarSolid } from '@admiral-ds/icons/build/system/StarSolid.svg';
import { T } from '#/components/T';
import { withDesign } from 'storybook-addon-designs';
import { Badge } from '#/components/Badge';

const Divider = styled.div`
  width: 10px;
`;

const Separator = styled.div`
  height: 20px;
`;

export default {
  title: 'Example/ButtonGroup',
  decorators: [withDesign],
  component: ButtonGroup,
  parameters: {
    design: [
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70132',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70273',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70157',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70180',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70203',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70134',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70226',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/HCiO63zg2hPSXTHuEdpRtG/Admiral-2.0-UI-Kit?node-id=7582%3A70249',
      },
    ],
  },
  argTypes: {
    dimension: {
      control: {
        type: 'radio',
        options: ['l', 'm', 's'],
      },
    },
  },
} as ComponentMeta<typeof ButtonGroup>;

const ButtonGroupDemo: ComponentStory<typeof ButtonGroup> = () => {
  const list = [
    { title: 'Active button', content: '1' },
    { title: 'Default button', content: '2' },
    { title: 'Disabled button', content: '3', disabled: true },
  ];
  return (
    <>
      <T font="Additional/L" as="div">
        Dimension - l
      </T>
      <Separator />
      <ButtonGroup dimension={'l'}>
        {list.map((item, index) => {
          return (
            <ButtonGroupItem key={index} active={index === 0} disabled={item.disabled}>
              {item.title}
            </ButtonGroupItem>
          );
        })}
      </ButtonGroup>
      <Separator />
      <ButtonGroup dimension={'l'} appearance="primary">
        {list.map((item, index) => {
          return (
            <ButtonGroupItem key={index} active={index === 0} disabled={item.disabled}>
              {item.title}
            </ButtonGroupItem>
          );
        })}
      </ButtonGroup>
      <Separator />
      <T font="Additional/L" as="div">
        Dimension - m
      </T>
      <Separator />
      <ButtonGroup dimension={'m'}>
        {list.map((item, index) => {
          return (
            <ButtonGroupItem key={index} active={index === 0} disabled={item.disabled}>
              {item.title}
            </ButtonGroupItem>
          );
        })}
      </ButtonGroup>
      <Separator />
      <T font="Additional/L" as="div">
        Dimension - s
      </T>
      <Separator />
      <ButtonGroup dimension={'s'}>
        {list.map((item, index) => {
          return (
            <ButtonGroupItem key={index} active={index === 0} disabled={item.disabled}>
              {item.title}
            </ButtonGroupItem>
          );
        })}
      </ButtonGroup>
    </>
  );
};

const ButtonGroupWithIconDemo = () => (
  <ButtonGroup dimension={'l'}>
    <ButtonGroupItem active>
      <StarSolid width={20} />
      <Divider />
      Active button
    </ButtonGroupItem>
    <ButtonGroupItem>Default button</ButtonGroupItem>
    <ButtonGroupItem disabled>
      Disabled button
      <Divider />
      <StarSolid width={17} />
    </ButtonGroupItem>
  </ButtonGroup>
);

const Template: ComponentStory<typeof ButtonGroup> = ({ dimension, className, ...props }) => {
  const [active, setActive] = useState(0);

  const list = [
    {
      title: (
        <>
          <StarSolid width={17} style={{ marginRight: 8 }} />
          Active button
        </>
      ),
    },
    {
      title: (
        <>
          Default button <Badge style={{ marginLeft: 8 }}>6</Badge>
        </>
      ),
    },
    { title: 'Disabled button', disabled: true },
  ];

  return (
    <ButtonGroup dimension={dimension} className={className} {...props}>
      {list.map((item, index) => (
        <ButtonGroupItem
          key={index}
          active={index === active}
          onClick={() => setActive(index)}
          disabled={item.disabled}
        >
          {item.title}
        </ButtonGroupItem>
      ))}
    </ButtonGroup>
  );
};

export const Example = ButtonGroupDemo.bind({});

export const ButtonGroupWithIcon = ButtonGroupWithIconDemo.bind({});
export const Playground = Template.bind({});
Playground.args = {};
