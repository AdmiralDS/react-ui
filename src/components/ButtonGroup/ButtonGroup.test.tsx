import { ButtonGroupItem } from '#src/components/ButtonGroup/ButtonGroupItem';
import { ButtonGroup } from '#src/components/ButtonGroup';
import { LIGHT_THEME } from '#src/components/themes';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

const BUTTON_TEXT = 'Button';
const BUTTON_CLICK_TEXT = 'Button Click';
const BUTTON_DISABLED_TEXT = 'Button Disabled';

describe('ButtonGroup', () => {
  it('should render component with Button', () => {
    const { container } = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <ButtonGroup dimension={'m'}>
          <ButtonGroupItem>{BUTTON_TEXT}</ButtonGroupItem>
        </ButtonGroup>
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render component with Buttons', () => {
    const { container } = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <ButtonGroup dimension={'m'}>
          <ButtonGroupItem>{BUTTON_TEXT}</ButtonGroupItem>
          <ButtonGroupItem>{BUTTON_TEXT}</ButtonGroupItem>
          <ButtonGroupItem>{BUTTON_TEXT}</ButtonGroupItem>
          <ButtonGroupItem>{BUTTON_TEXT}</ButtonGroupItem>
        </ButtonGroup>
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders the buttonGroup', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <ButtonGroup dimension={'m'}>
          <ButtonGroupItem>{BUTTON_TEXT}</ButtonGroupItem>
        </ButtonGroup>
      </ThemeProvider>,
    );
    expect(screen.getByRole('button')).toHaveTextContent(`${BUTTON_TEXT}`);
  });

  test('should call onClick when user clicks on last button', () => {
    const handleClick = jest.fn();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <ButtonGroup dimension={'m'}>
          <ButtonGroupItem onClick={handleClick}>{BUTTON_TEXT}</ButtonGroupItem>
          <ButtonGroupItem onClick={handleClick}>{BUTTON_CLICK_TEXT}</ButtonGroupItem>
        </ButtonGroup>
      </ThemeProvider>,
    );

    fireEvent.click(screen.getByText(`${BUTTON_CLICK_TEXT}`));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('Dont should call onClick when user clicks on last button', () => {
    const handleClick = jest.fn();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <ButtonGroup dimension={'m'}>
          <ButtonGroupItem onClick={handleClick}>{BUTTON_TEXT}</ButtonGroupItem>
          <ButtonGroupItem disabled onClick={handleClick}>
            {BUTTON_CLICK_TEXT}
          </ButtonGroupItem>
        </ButtonGroup>
      </ThemeProvider>,
    );

    fireEvent.click(screen.getByText(`${BUTTON_CLICK_TEXT}`));
    expect(handleClick).toHaveBeenCalledTimes(0);
  });

  test('should focus on component when user on press key tab', () => {
    const handleClick = jest.fn();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <ButtonGroup data-testid="test" dimension={'m'}>
          <ButtonGroupItem data-testid={BUTTON_CLICK_TEXT} onClick={handleClick}>
            {BUTTON_CLICK_TEXT}
          </ButtonGroupItem>
          <ButtonGroupItem data-testid={BUTTON_TEXT} onClick={handleClick}>
            {BUTTON_TEXT}
          </ButtonGroupItem>
          <ButtonGroupItem disabled onClick={handleClick}>
            {BUTTON_DISABLED_TEXT}
          </ButtonGroupItem>
        </ButtonGroup>
      </ThemeProvider>,
    );
    expect(document.body).toHaveFocus();
    userEvent.tab();
    expect(screen.getByTestId(BUTTON_CLICK_TEXT)).toHaveFocus();
    userEvent.tab();
    expect(screen.getByTestId(BUTTON_TEXT)).toHaveFocus();
    userEvent.tab();
    expect(document.body).toHaveFocus();
  });
});
