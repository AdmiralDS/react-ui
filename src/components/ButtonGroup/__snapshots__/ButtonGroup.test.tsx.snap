// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render component with Button 1`] = `
.c2 {
  border: none;
  border-radius: 22px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  outline: 0;
  background-color: transparent;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  white-space: nowrap;
  color: #626F84;
}

[data-appearance~='primary'] .c2.c2 {
  color: #0D69F2;
}

.c2:hover:not(:disabled) {
  background: #D3D8DE;
  color: #626F84;
}

.c2 *[fill^='#'] {
  fill: #626F84;
}

[data-appearance~='primary'] .c2.c2 *[fill^='#'] {
  fill: #0D69F2;
}

[data-appearance~='primary'] .c2.c2 {
  background-color: transparent;
}

[data-appearance~='primary'] .c2.c2:hover:not(:disabled) {
  background: #F1F2F4;
  color: #0A54C2;
}

.c0 {
  background: #E2E5E9;
  border: 2px solid #E2E5E9;
  border-radius: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 32px;
  padding: 2px;
}

.c0[data-appearance~='primary'] {
  border: 2px solid #0D69F2;
  background-color: transparent;
}

.c0 .c1 {
  height: 100%;
  padding: 0 16px;
  font-size: 14px;
  line-height: 20px;
}

.c0 > *:not(:first-child) {
  margin-left: 8px;
}

<div>
  <div
    class="c0"
    data-appearance="secondary"
    tabindex="-1"
  >
    <button
      class="c1 c2"
      type="button"
    >
      Button
    </button>
  </div>
</div>
`;

exports[`ButtonGroup should render component with Buttons 1`] = `
.c2 {
  border: none;
  border-radius: 22px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  outline: 0;
  background-color: transparent;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  white-space: nowrap;
  color: #626F84;
}

[data-appearance~='primary'] .c2.c2 {
  color: #0D69F2;
}

.c2:hover:not(:disabled) {
  background: #D3D8DE;
  color: #626F84;
}

.c2 *[fill^='#'] {
  fill: #626F84;
}

[data-appearance~='primary'] .c2.c2 *[fill^='#'] {
  fill: #0D69F2;
}

[data-appearance~='primary'] .c2.c2 {
  background-color: transparent;
}

[data-appearance~='primary'] .c2.c2:hover:not(:disabled) {
  background: #F1F2F4;
  color: #0A54C2;
}

.c0 {
  background: #E2E5E9;
  border: 2px solid #E2E5E9;
  border-radius: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 32px;
  padding: 2px;
}

.c0[data-appearance~='primary'] {
  border: 2px solid #0D69F2;
  background-color: transparent;
}

.c0 .c1 {
  height: 100%;
  padding: 0 16px;
  font-size: 14px;
  line-height: 20px;
}

.c0 > *:not(:first-child) {
  margin-left: 8px;
}

<div>
  <div
    class="c0"
    data-appearance="secondary"
    tabindex="-1"
  >
    <button
      class="c1 c2"
      type="button"
    >
      Button
    </button>
    <button
      class="c1 c2"
      type="button"
    >
      Button
    </button>
    <button
      class="c1 c2"
      type="button"
    >
      Button
    </button>
    <button
      class="c1 c2"
      type="button"
    >
      Button
    </button>
  </div>
</div>
`;
